### DISCUSSION TITLE
Energy import

### DISCUSSION THREAD

**Discussion by hofvanta** [patrickvorgers/Home-Assistant-Import-Energy-Data#60] [General]
HI, saw your github scripts. I have a question. I run parallel to my old raspberry a new one (RP5). Begin this year I took it into production after full support. How can I import the data from old HA to new HA for certain months? I tried to export SQL data from statistics (check on only metadata_id which I need and between timestamps), but it will mesh up my HA DB seems like. How can you fix this?  Do you have any idea?

**Comment by patrickvorgers** [comment]
Probably the metadata_id of the different entities have changed and the sum column is not correct anymore. Therefore your data is not correct. My importscript corrects the sum according to the imported data.
You could use the "Home Assistant" datasource and export individual entities. This would be time consuming but could be workable for the energy sensors.

**Reply by hofvanta** [reply]
Thanks for answering. Can you guide me to the direction how? I dont need all sensors. Only the ones that are related and important in for the energy dashboard. The sum sql query output of the old HA (for each hours every day for three months)  should be sufficient right? The counters should not change. Or do I see it wrong?

**Reply by patrickvorgers** [reply]
Just follow the How-to in the [Home Assistant datasource section](https://github.com/patrickvorgers/Home-Assistant-Import-Energy-Data/tree/main/Datasources/Home%20Assistant). You have to use the scenario (top one) where an existing Home Assistant installation is available (your old Home Assistant installation).

This exports the entity data from your old Home Assistant setup and prepares it so that it can be imported using the generic script. The generic script handles all the recalculation regarding `sum` etc.
